packages/mui-material/src/SpeedDial/SpeedDial.d.ts

```diff
@@ -5,12 +5,53 @@ import { InternalStandardProps as StandardProps } from '..';
 import { FabProps } from '../Fab';
 import { TransitionProps } from '../transitions';
 import { SpeedDialClasses } from './speedDialClasses';
+import { CreateSlotsAndSlotProps, SlotProps } from '../utils/types';
 
 export type CloseReason = 'toggle' | 'blur' | 'mouseLeave' | 'escapeKeyDown';
 export type OpenReason = 'toggle' | 'focus' | 'mouseEnter';
 
+export interface SpeedDialSlots {
+  /**
+   * The component that renders the root.
+   * @default 'div'
+   */
+  root?: React.ElementType;
+  /**
+   * The component that renders the transition.
+   * [Follow this guide](/material-ui/speedDial/#transitioncomponent-prop) to learn more about the requirements for this component.
+   * @default {}
+   */
+  transition?: React.JSXElementConstructor<
+    TransitionProps & { children: React.ReactElement<any, any> }
+  >;
+}
+
+export interface SpeedDialComponentsPropsOverrides {}
+
+export interface SpeedDialTransitionSlotPropsOverrides {}
+
+export type SpeedDialSlotsAndSlotProps = CreateSlotsAndSlotProps<
+  SpeedDialSlots,
+  {
+    root: SlotProps<
+      React.ElementType<HTMLDivElement>,
+      SpeedDialComponentsPropsOverrides,
+      SpeedDialOwnerState
+    >;
+    transition: SlotProps<
+      React.JSXElementConstructor<TransitionProps>,
+      SpeedDialTransitionSlotPropsOverrides,
```

Comment:
Let's pass `{}` here and remove `SpeedDialTransitionSlotPropsOverrides` altogether. No need to add it at this moment.

```suggestion
     {},
```